x/marker/keeper/msg_server_test.go

```diff
@@ -0,0 +1,720 @@
+			expErr: "decoding bech32 failed: invalid separator index -1",
+		},
+		{
+			name: "should succeed to add to deny list",
+			msg:  types.MsgUpdateSendDenyListRequest{Denom: rMarkerDenom, Authority: authUser.String(), RemoveDeniedAddresses: []string{}, AddDeniedAddresses: []string{denyAddrToAdd.String()}},
+		},
+		{
+			name: "should succeed to remove from deny list",
+			msg:  types.MsgUpdateSendDenyListRequest{Denom: rMarkerDenom, Authority: authUser.String(), RemoveDeniedAddresses: []string{denyAddrToRemove.String()}, AddDeniedAddresses: []string{}},
+		},
+		{
+			name: "should succeed gov allowed for marker",
+			msg:  types.MsgUpdateSendDenyListRequest{Denom: rMarkerGovDenom, Authority: authority.String(), RemoveDeniedAddresses: []string{}, AddDeniedAddresses: []string{denyAddrToAddGov.String()}},
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(tc.name, func() {
+			res, err := s.msgServer.UpdateSendDenyList(sdk.WrapSDKContext(s.ctx),
+				&tc.msg)
+
+			if len(tc.expErr) > 0 {
+				s.Assert().Nil(res)
+				s.Assert().EqualError(err, tc.expErr)
+
+			} else {
+				s.Assert().NoError(err)
+				s.Assert().Equal(res, &types.MsgUpdateSendDenyListResponse{})
+			}
+		})
+	}
+}
+
+func (s *MsgServerTestSuite) TestAddNetAssetValue() {
+	authUser := testUserAddress("test")
+	notAuthUser := testUserAddress("blah")
+
+	markerDenom := "jackthecat"
+	markerAcct := authtypes.NewBaseAccount(types.MustGetMarkerAddress(markerDenom), nil, 0, 0)
+	s.app.MarkerKeeper.SetMarker(s.ctx, types.NewMarkerAccount(markerAcct, sdk.NewInt64Coin(markerDenom, 1000), authUser, []types.AccessGrant{{Address: authUser.String(), Permissions: []types.Access{types.Access_Transfer}}}, types.StatusProposed, types.MarkerType_RestrictedCoin, true, false, false, []string{}))
+
+	valueAcct := authtypes.NewBaseAccount(types.MustGetMarkerAddress(types.UsdDenom), nil, 0, 0)
+	s.app.MarkerKeeper.SetMarker(s.ctx, types.NewMarkerAccount(valueAcct, sdk.NewInt64Coin(types.UsdDenom, 1000), authUser, []types.AccessGrant{{Address: authUser.String(), Permissions: []types.Access{types.Access_Transfer}}}, types.StatusProposed, types.MarkerType_RestrictedCoin, true, false, false, []string{}))
+
+	finalizedMarkerDenom := "finalizedjackthecat"
+	finalizedMarkerAcct := authtypes.NewBaseAccount(types.MustGetMarkerAddress(finalizedMarkerDenom), nil, 1, 0)
+	s.app.MarkerKeeper.SetMarker(s.ctx, types.NewMarkerAccount(finalizedMarkerAcct, sdk.NewInt64Coin(finalizedMarkerDenom, 1000), authUser, []types.AccessGrant{{Address: authUser.String(), Permissions: []types.Access{types.Access_Transfer}}}, types.StatusFinalized, types.MarkerType_RestrictedCoin, true, false, false, []string{}))
+
+	testCases := []struct {
+		name   string
+		msg    types.MsgAddNetAssetValuesRequest
+		expErr string
+	}{
+		{
+			name: "no marker found",
```

Comment:
This is related to my previous comment. You mentioned PENDING state in a comment, if you do end up adding logic for it then you may want to add a test for it.

Reply:
Yeah, old design.  Good eye though!